rpc/src/client.rs

```diff
@@ -188,14 +182,85 @@ impl Client {
 
 pub struct ClientAsync {
     connections: DashMap<String, Arc<ClientConnectionAsync>>,
-    queue: CircularQueue,
+    queue: Arc<CircularQueue>,
 }
 
 impl Default for ClientAsync {
     fn default() -> Self {
+        let mut queue = CircularQueue::new();
+        queue.init();
         Self {
             connections: DashMap::new(),
-            queue: CircularQueue::new(),
+            queue: Arc::new(queue),
+        }
+    }
+}
+
+// parse_response
+// try to get response from sequence of connections and write to callbacks
+pub async fn parse_response(
+    mut read_stream: OwnedReadHalf,
+    connection: Arc<ClientConnectionAsync>,
+    queue: Arc<CircularQueue>,
+) {
+    loop {
```

Comment:
I think need to `break loop`, after `remove_connection`, maybe change the `status` of the `connection` is nice

Reply:
Yes! I'm not sure shell we use a lock?

Reply:
The `status` of `connection` already use `RwLock`

Reply:
done